src/main/scala/is/hail/utils/Graph.scala

```diff
@@ -28,6 +33,59 @@ object Graph {
     m
   }
 
+  def maximalIndependentSet(edges: Array[Row], edgeType: Type, tieBreaker: Option[String]): Set[Any] = {
+    val edges2 = edges.map { r =>
+      val Row(x, y) = r
+      (x, y)
+    }
+
+    if (edges2.length > 400000)
+      warn(s"over 400,000 edges are in the graph; maximal_independent_set may run out of memory")
+
+    val wrappedEdgeType = TTuple(edgeType)
```

Comment:
Can you explain the reason for making singleton tuples? Also, should `edgeType` really be `nodeType`, or am I misunderstanding?

Reply:
Ah, I get it now. Thanks.